### ISSUE TITLE
Warning: React does not recognize the `propsName` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `propsname` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

### ISSUE THREAD

**Issue by knposhti** [styled-components/styled-components#4232]
When I try to migrate styled-component to v6, I get this warning in the console.
Warning: React does not recognize the `propsName` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `propsname` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

any suggestions or solutions?

**Comment by quantizor** [comment]
That appears to be a custom prop you are using, we don't have that in our library.

**Comment by knposhti** [comment]
@quantizor,
correct, those are the custom props and I want to add them on my styled component but it gives me this warning from styled-component.
example is as below,


```
interface InputItemProps {
  isDisabled?: boolean;
}

const InputDiv = styled.div<InputItemProps>`
  display: inline-block;
  position: relative;
  width: 100%;
  color: ${(props) =>
    props.isDisabled
      ? "rgb(141, 141, 149)"
      : "rgb(23, 28, 67)";
`;

<InputDiv isDisabled={true} />
```

**Comment by quantizor** [comment]
Use transient prop syntax instead to avoid that warning https://styled-components.com/docs/api#transient-props
